lib/components/form/call-taker/date-time-picker.tsx

```diff
@@ -197,42 +190,64 @@ const DateTimeOptions = ({
         })
       })
     }
-
-    if (
-      syncSortWithDepartArrive &&
-      DepartArriveTypeMap[departArrive] !== sort.type
-    ) {
-      importedUpdateItineraryFilter({
-        sort: {
-          ...sort,
-          type: DepartArriveTypeMap[departArrive]
-        }
-      })
-    }
   }, [dateTime, departArrive, homeTimezone, setQueryParam])
 
-  // Handler for updating the time and date fields when NOW is selected
-  useEffect(() => {
-    if (departArrive === 'NOW') {
-      setTime(getCurrentTime(homeTimezone))
-      setDate(getCurrentDate(homeTimezone))
-      setTypedTime(
-        safeFormat(dateTime, timeFormat, {
-          timeZone: homeTimezone
+  const handleDepartArriveChange = useCallback(
+    (e: ChangeEvent<HTMLSelectElement>) => {
+      const newValue = e.target.value as DepartArriveValue
+      setDepartArrive(newValue)
+
+      // Handler for updating the time and date fields when NOW is selected
+      if (newValue === 'NOW') {
+        handleTimeChange(getCurrentTime(homeTimezone))
+        setDate(getCurrentDate(homeTimezone))
+        setTypedTime(
+          safeFormat(dateTime, timeFormat, {
+            timeZone: homeTimezone
+          })
+        )
+      }
+
+      // Update sort type if needed
+      if (
+        syncSortWithDepartArrive &&
+        DepartArriveTypeMap[newValue] !== sort.type
+      ) {
+        importedUpdateItineraryFilter({
+          sort: {
+            ...sort,
+            type: DepartArriveTypeMap[newValue]
+          }
         })
-      )
-    }
-  }, [departArrive, setTime, setDate, homeTimezone])
+      }
+    },
+    [syncSortWithDepartArrive, sort, importedUpdateItineraryFilter]
```

Comment:
Here, eslint is suggesting to add `dateTime`, `handleTimeChange`, `homeTimezone` to the dependency array.

Reply:
Good note, added them.